src/EventHandlers/CLPool/CLPoolBurnLogic.ts

```diff
@@ -1,49 +1,42 @@
-import type { CLPool_Burn_event, Token, handlerContext } from "generated";
-import { updateReserveTokenData } from "../../Helpers";
+import type { CLPool_Burn_event, Token } from "generated";
+import { calculateTotalLiquidityUSD } from "../../Helpers";
 
 export interface CLPoolBurnResult {
   liquidityPoolDiff: {
     reserve0: bigint;
     reserve1: bigint;
     totalLiquidityUSD: bigint;
-    lastUpdatedTimestamp: Date;
   };
   userLiquidityDiff: {
-    netLiquidityAddedUSD: bigint;
+    currentLiquidityUSD: bigint;
     currentLiquidityToken0: bigint;
     currentLiquidityToken1: bigint;
-    timestamp: Date;
   };
 }
 
-export async function processCLPoolBurn(
+export function processCLPoolBurn(
   event: CLPool_Burn_event,
   token0Instance: Token,
   token1Instance: Token,
-  context: handlerContext,
-): Promise<CLPoolBurnResult> {
-  // Update reserve data using the same approach as Pool events
-  const reserveData = await updateReserveTokenData(
-    token0Instance,
-    token1Instance,
+): CLPoolBurnResult {
+  // Calculate USD values using already-refreshed token prices from loadPoolData
+  const totalLiquidityUSD = calculateTotalLiquidityUSD(
     event.params.amount0,
     event.params.amount1,
-    event,
-    context,
+    token0Instance,
+    token1Instance,
   );
 
   const liquidityPoolDiff = {
-    reserve0: event.params.amount0,
-    reserve1: event.params.amount1,
-    totalLiquidityUSD: reserveData.totalLiquidityUSD,
-    lastUpdatedTimestamp: new Date(event.block.timestamp * 1000),
+    reserve0: -event.params.amount0,
+    reserve1: -event.params.amount1,
+    totalLiquidityUSD: -totalLiquidityUSD,
   };
 
   const userLiquidityDiff = {
-    netLiquidityAddedUSD: -reserveData.totalLiquidityUSD, // Negative for burn (removal)
+    currentLiquidityUSD: -totalLiquidityUSD, // Negative for burn (removal)
```

Comment:
If this value represents the change in liquidity, name `currentLiquidityUSD` may be misleading 

`netLiquidityAddedUSD` may be a better name